### PR TITLE
st_tree: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **st_tree/1.2.1**

st_tree is stl like container for tree data structures.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/st_tree/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "st_tree"
```
Comment (uilianries) [inline recipes/st_tree/all/conanfile.py:42]: correct https://github.com/erikerlandson/st_tree/blob/develop/CMakeLists.txt#L40 :+1:

File: recipes/st_tree/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+required_conan_version = ">=1.33.0"
+
+class STTreeConan(ConanFile):
+    name = "st_tree"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A fast and flexible c++ template class for tree data structures"
+    topics = ("stl", "container", "data-structures")
+    homepage = "https://github.com/erikerlandson/st_tree"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/st_tree/all/conanfile.py:15]:
```suggestion
```

As you don't have a cmake wrapper which would consume conanbuildinfo.cmake, you don't need any generator. But it doesn't harm

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/st_tree/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/st_tree/all/conanfile.py:15]: I agree with you.
I forget this recipe is header only.

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**